x-pack/plugins/observability_solution/apm/server/routes/entities/services/routes.ts

```diff
@@ -46,6 +58,98 @@ const servicesEntitiesRoute = createApmServerRoute({
   },
 });
 
+const servicesEntitiesDetailedStatisticsRoute = createApmServerRoute({
+  endpoint: 'POST /internal/apm/entities/services/detailed_statistics',
+  params: t.type({
+    query: t.intersection([
+      environmentRt,
+      kueryRt,
+      rangeRt,
+      t.intersection([offsetRt, probabilityRt, serviceTransactionDataSourceRt]),
+      t.type({
+        bucketSizeInSeconds: toNumberRt,
+      }),
+    ]),
+    body: t.type({ serviceNames: jsonRt.pipe(t.array(t.string)) }),
+  }),
+  options: { tags: ['access:apm'] },
+  handler: async (resources) => {
+    const {
+      context,
+      params,
+      request,
+      plugins: { security, logsDataAccess },
+    } = resources;
+
+    const [coreContext, logsDataAccessStart] = await Promise.all([
+      context.core,
+      logsDataAccess.start(),
+    ]);
+
+    const {
+      environment,
+      kuery,
+      offset,
+      start,
+      end,
+      probability,
+      documentType,
+      rollupInterval,
+      bucketSizeInSeconds,
+    } = params.query;
+
+    const { serviceNames } = params.body;
+
+    const [apmEventClient, randomSampler] = await Promise.all([
+      getApmEventClient(resources),
+      getRandomSampler({ security, request, probability }),
+    ]);
+
+    if (!serviceNames.length) {
+      throw Boom.badRequest(`serviceNames cannot be empty`);
+    }
+
+    const logsParams = {
+      esClient: coreContext.elasticsearch.client.asCurrentUser,
+      identifyingMetadata: 'service.name',
+      timeFrom: start,
+      timeTo: end,
+      kuery,
+      serviceEnvironmentQuery: environmentQuery(environment),
+      serviceNames,
+    };
+
+    const currentPeriodLogsRateTimeseries =
```

Comment:
it would be better to use 
```
Promise.all([
     logsDataAccessStart.services.getLogsRateTimeseries(logsParams);
     logsDataAccessStart.services.getLogsErrorRateTimeseries(logsParams);
getServiceTransactionDetailedStatsPeriods({
        environment,
        kuery,
        apmEventClient,
        documentType,
        rollupInterval,
        bucketSizeInSeconds,
        offset,
        serviceNames,
        start,
        end,
        randomSampler,
      });
    ]);
```

Reply:
changed, thanks 👍